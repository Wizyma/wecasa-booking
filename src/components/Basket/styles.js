import styled from 'styled-components';

export const FixedWrapper = styled.div`
  position: fixed;
  bottom: 10px;
  right: 10px;
  background-color: #0069c0;
  color: white;
  width: 60px;
  height: 60px;
  border-radius: 50px;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
  -webkit-transition: width 2s, height 2s, border-radius 2s; /* For Safari 3.1 to 6.0 */
  transition: width 2s, height 2s, border-radius 2s;
  :hover {
    width: ${props => props.basket && '200px'};
    height: ${props => props.basket && '200px'};
    border-radius: ${props => props.basket && '5px'};
  }
`;
