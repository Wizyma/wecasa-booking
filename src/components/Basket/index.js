import React, { useContext, memo, useState } from 'react';

import { FixedWrapper } from './styles';
import Store from '../../context';

const renderBasket = basket => <div>{basket.length}</div>;
//     <p>Prix total: {basket.map(b => b.price).reduce((a, b) => a + b, 0)}</p>

const renderBasketItems = (basket, call) =>
  basket ? (
    <div>
      <p>Prix total: {basket.map(b => b.price).reduce((a, b) => parseInt(a, 10) + parseInt(b, 10), 0)}</p>
      <p>Durée éstimé: {basket.map(b => b.duration).reduce((a, b) => a + b, 0)}</p>
      <button type="button" onClick={() => call({ type: 'RESET_BASKET' })}>
        Vider le panier
      </button>
      <button type="button">
        Voir le panier
        {/* this should be done using react router */}
      </button>
    </div>
  ) : (
    0
  );

const Basket = () => {
  const [store, call] = useContext(Store);
  const [showItems, setState] = useState(false);

  return (
    <FixedWrapper
      onMouseEnter={() => setState(!showItems)}
      onMouseLeave={() => setState(!showItems)}
      basket={store.basket}
    >
      {!showItems ? (
        store.basket ? (
          renderBasket(store.basket)
        ) : (
          <div>0</div>
        )
      ) : (
        renderBasketItems(store.basket, call)
      )}
    </FixedWrapper>
  );
};

export default memo(Basket);
