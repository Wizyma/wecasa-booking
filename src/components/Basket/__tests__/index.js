import { render, cleanup } from 'react-testing-library';
import React from 'react';

import Basket from '../';
import Store from '../../../context';

afterEach(cleanup);

describe('Basket', () => {
  it('should render Basket with mocked data', () => {
    const mockedData = {
      bakest: [
        {
          title: 'Shampoing',
          duration: 10,
          price: 500,
        },
        {
          title: 'Shampoing',
          duration: 10,
          price: 500,
        },
      ],
    };

    const tree = (
      <Store.Provider value={[mockedData]}>
        <Basket />
      </Store.Provider>
    );
    const { baseElement } = render(tree);

    expect(baseElement).toMatchSnapshot();
  });

  it('should return 0 when basket is empty', () => {
    const tree = (
      <Store.Provider value={[{}]}>
        <Basket />
      </Store.Provider>
    );
    const { baseElement } = render(tree);

    expect(baseElement).toMatchSnapshot();
  });
});
