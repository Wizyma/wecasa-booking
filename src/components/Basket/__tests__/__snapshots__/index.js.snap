// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket should render Basket with mocked data 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa fKXRkp"
    >
      <div>
        0
      </div>
    </div>
  </div>
</body>
`;

exports[`Basket should return 0 when basket is empty 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa fKXRkp"
    >
      <div>
        0
      </div>
    </div>
  </div>
</body>
`;
